pkg/registry/nsmgr/server.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package nsmgr - provide Network Service Mesh manager registry tracker, to update endpoint registrations.
+package nsmgr
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/registry"
+
+	"github.com/networkservicemesh/sdk/pkg/registry/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/serialize"
+)
+
+type nsmRegistryServer struct {
+	registry.NetworkServiceRegistryServer
+	serialize.Executor
+	manager *registry.NetworkServiceManager
+}
+
+// NewServer - construct a new NSM aware NSE registry server
+// Server is using latest NSM registration and update endpoints accordingly.
+func NewServer(manager *registry.NetworkServiceManager) registry.NetworkServiceRegistryServer {
+	return &nsmRegistryServer{
+		manager: manager,
```

Comment:
Do we allow the nil manager here?

Reply:
NewServer is not return error, so I could not return error here. But I think manager should be passed.